adal/src/main/java/com/microsoft/aad/adal/AuthenticationActivity.java

```diff
@@ -888,13 +891,37 @@ public void alias(final String alias) {
                             request.cancel();
                         }
                     },
-                    request.getKeyTypes(),
+                    mapKeyTypes(request.getKeyTypes()),
                     request.getPrincipals(),
                     request.getHost(),
                     request.getPort(),
                     null
             );
         }
+
+        /**
+         * Map instances of key types without a literal reference in {@link KeyProperties} to corresponding constants in KeyProperties.
+         * @param keyTypes array of key types.
+         * @return array of mapped key types.
+         */
+        @RequiresApi(api = Build.VERSION_CODES.M)
+        @Nullable
+        public String[] mapKeyTypes(@Nullable final String[] keyTypes) {
```

Comment:
maybe you can put this code in a static method in common and use it in both places.

Reply:
Made mapKeyTypes in common static: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/2015/commits/19f0d0f81c9cc2f2c0218451e862e3dbcbe8c6b5